src/test/java/com/xceptance/neodymium/util/SelenideAddonsTest.java

```diff
@@ -640,4 +643,108 @@ public void testOpenHtmlContentWithCurrentWebDriver()
         SelenideAddons.openHtmlContentWithCurrentWebDriver(textHtml);
         Assert.assertEquals(text, $("body").getText());
     }
+
+    private int customWaitingTime = 3000;
+
+    @Test
+    public void testOptionalWaitUntil()
+    {
+        openBlogPage();
+
+        // wait until the optional privacy dialog is visible
+        SelenideElement privaceDialog = $("#privacy-message");
+        boolean isVisible = SelenideAddons.optionalWaitUntilCondition(privaceDialog, visible, customWaitingTime);
+        assertTrue("the privacy message dialog was not found within the timeframe", isVisible);
+
+        // try wait until the optional privacy dialog is hidden, which will fail
+        long startTime = new Date().getTime();
+        boolean isHidden = SelenideAddons.optionalWaitUntilCondition(privaceDialog, hidden, customWaitingTime);
+        long endTime = new Date().getTime();
+
+        // check that the is false as is expected
+        assertFalse("the privacy message dialog was unexpectedly hidden during the timeframe", isHidden);
+
+        // check that runtime of the wait until method was as long as expected
+        int waitingTime = customWaitingTime * Neodymium.configuration().optionalElementRetryCount();
+        Assert.assertTrue("Runtime was shorter than expected", Range.between(waitingTime, waitingTime + customWaitingTime)
```

Comment:
Please adjust the error message to something like 
`Runtime was not within the expected range`
since the runtime can also be longer.

Reply:
Please refactor this into one private validation method in order to avoid code duplication throughout the 4 test cases. Try to perform the same validation. The other test cases also check with `Range.contains`